src/main/java/org/rumbledb/api/Item.java

```diff
@@ -109,96 +119,210 @@ public int compareTo(Item other) {
+     * 
+     * @param key a key.
+     * @return the value associated with key.
+     */
+    public Item getItemByKey(String key) {
         throw new RuntimeException("Item is not an object.");
     }
 
-    public void putItemByKey(String s, Item value) {
+    /**
+     * Adds a key-value pair, if it is an object item.
+     * 
+     * @param key a key.
+     * @param value a value.
+     */
+    public void putItemByKey(String key, Item value) {
         throw new RuntimeException("Item is not an object.");
     }
 
+    /**
+     * Returns the size of the item, if it is an array item.
+     * 
+     * @return the size as an int.
+     */
     public int getSize() {
         throw new RuntimeException("Item is not an array.");
     }
 
+    /**
+     * Returns the string value of the item, if it is a atomic item of type string.
+     * 
+     * @return the string value.
+     */
     public String getStringValue() {
         throw new RuntimeException("Item is not a string.");
     }
 
+    /**
+     * Returns the boolean value of the item, if it is a atomic item of type boolean.
+     * 
+     * @return the boolean value.
+     */
     public boolean getBooleanValue() {
         throw new RuntimeException("Item is not a boolean.");
     }
 
+    /**
+     * Returns the double value of the item, if it is a atomic item of type double.
+     * 
+     * @return the double value.
+     */
     public double getDoubleValue() {
         throw new RuntimeException("Item is not a double.");
     }
 
+    /**
+     * Returns the integer value of the item, if it is a atomic item of type integer.
+     * 
+     * @return the integer value as an int.
+     */
     public int getIntegerValue() {
         throw new RuntimeException("Item is not an integer.");
     }
 
-    public BigDecimal getDecimalValue() {
+	/**
+	 * Returns the decimal value of the item, if it is a atomic item of type decimal.
+	 * 
+	 * @return the decimal value as a BigDecimal.
+	 */
+	public BigDecimal getDecimalValue() {
         throw new RuntimeException("Item is not a big decimal.");
     }
 
+	/**
+	 * Please do not use, item type API not publicly released yet.
+	 * @param type an ItemType.
+	 * @return true if it matches the item type.
+	 */
     public abstract boolean isTypeOf(ItemType type);
 
-    public boolean isArray() {
+    /**
+	 * Tests whether the item is an array.
+	 * 
+	 * @return true if it is an array, false otherwise.
+	 */
+	public boolean isArray() {
         return false;
     }
 
+    /**
+	 * Tests whether the item is an array.
+	 * 
+	 * @return true if it is an array, false otherwise.
+	 */
     public boolean isObject() {
         return false;
     }
 
+    /**
+	 * Tests whether the item is an atomic item.
+	 * 
+	 * @return true if it is an atomic item, false otherwise.
+	 */
     public boolean isAtomic() {
         return false;
     }
 
-    public boolean isNumber() {
-        return false;
-    }
-
+    /**
```

Comment:
minor issue with tabs appearing the javadoc

Reply:
Good catch, this is fixed.